oauthproxy_test.go

```diff
@@ -1786,6 +1786,124 @@ func TestGetRedirect(t *testing.T) {
 			assert.Equal(t, tt.expectedRedirect, redirect)
 		})
 	}
+
+	proxyOpts := baseTestOptions()
+	proxyOpts.ReverseProxy = true
+	proxyOpts.WhitelistDomains = append(proxyOpts.WhitelistDomains, ".example.com")
+	proxyOpts.WhitelistDomains = append(proxyOpts.WhitelistDomains, ".example.com:8443")
+	err = validation.Validate(proxyOpts)
+	assert.NoError(t, err)
+	require.NotEmpty(t, proxyOpts.ProxyPrefix)
+	proxy, err = NewOAuthProxy(proxyOpts, func(s string) bool { return false })
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	proxyTests := []struct {
+		name             string
+		url              string
+		proto            string
+		host             string
+		uri              string
+		expectedRedirect string
+	}{
+		{
+			name:             "proxied request outside of ProxyPrefix redirects to proxiexd URL",
+			url:              "https://oauth.example.com/foo/bar",
```

Comment:
I think this would be cleaner and allow you to switch to more complex test cases if you replaced `url`, `proto`, `host` that you direct set to their `X-Forwarded-*` peers below with a `headers map[string]string`.

This way you can better test how these headers work when intermingled with other redirect directing headers (nginx's `X-Auth-Request-Redirect` particularly)

Reply:
Then in your subtest setup:

```
for header, value := range tt.headers {
    req.Header.Add(header, value)
}
```